app/models/order.rb

```diff
@@ -0,0 +1,51 @@
+class Order < ActiveRecord::Base
+  has_many :order_items
+
+  validates :status, presence: true
+  [:email, :mailing_address, :zip, :name_on_card, :last_four, :card_exp].each do |attribute|
+    validates attribute, presence: true, on: :update
+  end
+  validates :last_four, numericality: { only_integer: true }, on: :update
+  validates :zip, length: { is: 5 }, on: :update
+  validate :card_exp_cannot_be_in_the_past
+
+  # validate :still_in_stock, on: :update
+  #
+  # def still_in_stock
+  #   errors.add(:stock, "Some items have gone out of stock") unless Order.find(cookies.signed[:order]).instock
+  # end
+
+  def card_exp_cannot_be_in_the_past
+    if card_exp.present? && card_exp < Date.today
+      errors.add(:exp, "Can't be in the past")
+    end
+  end
+
+  def total(user_id)
+    revenue = 0
+    order_items.each do |oi|
+      if oi.product.user.id == user_id
+        revenue += (oi.product.price * oi.quantity)
+      end
+    end
+    return revenue
+  end
```

Comment:
Since you're only ever calling this method in the views, you could define it as a helper method inside `app/helpers/application_helper.rb` to dry up your models and separate responsibility.